### DISCUSSION TITLE
If Atom Is Considered Dead or Dying, What's the Best Alternative?

### DISCUSSION THREAD

**Discussion by barmanroys** [atom/atom#23474] [Q&A]
From some of the discussions, I could not help but notice (with a tinge of sadness) that a good number of users and contributors think atom is on life support. As a Linux (Ubuntu) user, this IDE is something I came to love and appreciate. As I ponder over what is the best alternative, I wonder why not ask the fellow Atom users who would know best. 

My basic criteria and user scenarios are

- Most frequently used languages: Python, C++ and Bash, and a lot of config files, json etc. because I work with docker, kubernets. 
- Free and open source
- Great support for dark mode
- Basic functionalities like autocomplete, showing interactive variable lists, call hierarchy, Linting and formatting on save, live method documentations etc.
- Extensively compatible with language server protocols to enforce formatters like Clang, Black, PEP 8 etc.

Some of these, such as support for Dark mode and Python support are ruling out Code::Block. Potentially, a close competitor is VSCode, but somehow just want to avoid it, not to say I never used it at all. 

Also, I am not a real terminal Ninja to work with Vim or Emacs. Although I am sure they can be learnt, right now I want to avoid the curve to focus on other stuffs. Atom seemed like a perfect sweet spot with its customisability and ease of use, but curious what else is out there (as backup) for my user scenarios?

**Comment by mauricioszabo** [comment]
To be honest, none. Yes, atom _is_ on life support, unless the community can step up and keep the editor, which I don't see happening soon (or at all).

I tried to hack around solutions and old bugs I kept having, but I was not even able to reliably compile atom, much less upgrade to latest electron versions and start to develop enhancements. I really can't speak for alternatives, as they are all WAY less hackable (and I can't recommend VSCode at all. Maybe VSCodium, but even then I would take with a caution - the APIs are made to be limited, and the code is convoluted maybe to avoid a user "hacking" into internals).

What I though about doing is hacking some internals in ClojureScript (because of its fast feedback cycle) but it's honestly too much work to be doing alone...

**Comment by UziTech** [comment]
> unless the community can step up and keep the editor

There is a community version of atom at https://github.com/atom-community/atom/ it's support is very much come and go but if you are able to build it yourself it is a great alternative.

**Reply by mauricioszabo** [reply]
Neither `atom/atom` or `atom-community/atom` install with newest node. Also, their discord servers are almost 100% silent, and one of the maintainers said himself that the project is kinda dead too...

Both implementations are in what people call "maintenance mode". Both say that the "core" is in good shape, despite multiple bugs not being addressed, the text editor needing some love, no semantic highlight/LSP support out of the box/outdated and slow libraries in the UI... in fact, we can't upgrade to newest electron because the editor uses outdated APIs, and there are some issues with the file watcher npm module... so unless Atom is easier for people to hack, I don't see it surviving in the future, unfortunately, unless GitHub somehow decides to spend time on it, which I don't think they'll do anyway.

**Reply by icecream17** [reply]
Where is the atom discord?

**Reply by mauricioszabo** [reply]
It's the atom-community discord server: https://discord.gg/pqcCrq28. But as of now, it's 100% silent too...

**Reply by ChefDrew** [reply]
Hello,
 I am **brand new** to the world of code and mac. I am trying to switch careers from the kitchen to software. Im enjoying the code part. Please what beyond atom would be good for me to use and or how do I learn atom. Thank you for reading. I apologize to interrupt your feed.
